vic/drivers/image/src/get_force_type.c

```diff
@@ -0,0 +1,148 @@
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *****************************************************************************/
+
+#include <vic_def.h>
+#include <vic_run.h>
+#include <vic_driver_image.h>
+
+/******************************************************************************
+ * @brief    This routine determines the current forcing file data type and
+ *           stores its location in the description of the current forcing file.
+ *****************************************************************************/
+void
+get_force_type(char *cmdstr,
+               int   file_num,
+               int  *field)
+{
+    extern param_set_struct param_set;
+
+    char                    optstr[MAXSTRING];
+    char                    flgstr[MAXSTRING];
+    char                    ncvarname[MAXSTRING];
+    int                     type;
+
+    type = SKIP;
+    strcpy(ncvarname, "MISSING");
+
+    /** Initialize flgstr **/
+    strcpy(flgstr, "NULL");
+
+    if ((*field) >= (int) param_set.N_TYPES[file_num]) {
+        log_err("Too many variables defined for forcing file %i., was "
+                "expecting at most %zu and got %d", file_num + 1,
+                param_set.N_TYPES[file_num], *field);
+    }
+
+    sscanf(cmdstr, "%*s %s %s", optstr, ncvarname);
+
+    /***************************************
+       Get meteorological data forcing info
+    ***************************************/
+
+    /* type 0: air temperature [C] */
+    if (strcasecmp("AIR_TEMP", optstr) == 0) {
+        type = AIR_TEMP;
+    }
+    /* type 1: albedo [fraction] */
+    else if (strcasecmp("ALBEDO", optstr) == 0) {
+        type = ALBEDO;
+    }
+    /* type 2: atmospheric CO2 mixing ratio [ppm] */
+    else if (strcasecmp("CATM", optstr) == 0) {
+        type = CATM;
+    }
+    /* type 3: incoming channel flow [m3] */
+    else if (strcasecmp("CHANNEL_IN", optstr) == 0) {
+        type = CHANNEL_IN;
+    }
+    /* type 4: direct fraction of shortwave [fraction] */
+    else if (strcasecmp("FDIR", optstr) == 0) {
+        type = FDIR;
+    }
+    /* type 5: LAI [m2/m2] */
+    else if (strcasecmp("LAI_IN", optstr) == 0) {
+        type = LAI_IN;
+    }
+    /* type 6: incoming longwave radiation [W/m2] */
+    else if (strcasecmp("LONGWAVE",
+                        optstr) == 0 || strcasecmp("LWDOWN", optstr) == 0) {
+        type = LONGWAVE;
```

Comment:
I also prefer LWDOWN.  I think I added that check to allow flexibility but I suspect nobody uses LWDOWN, so probably we should standardize on LONGWAVE for backwards compatibility.  Or, if you prefer LWDOWN, switching to that is fine with me (we'd want to change SHORTWAVE to SWDOWN as well).  You'd also want to make sure output variable names are consistent: OUT_LONGWAVE would need to be OUT_LWDOWN, etc.  Then OUT_NET_LONG could become OUT_LWNET.  ETc.